src/components/Card/Card.js

```diff
@@ -0,0 +1,88 @@
+import React, { useReducer } from 'react'
+import { useCart } from "../../context/CartContext"
+import { useAuth } from "../../context/AuthenticationContext"
+import { Link, useNavigate } from "react-router-dom";
+import { addItemInWishlist, updateCart, removeItemFromWishlist } from '../../utilities/helpers/http-helper';
+
+const Card = (props) => {
+
+    const { authState: { token, isLogin } } = useAuth();
+    const data = props.item
+    const navigate = useNavigate()
+    const { cartState: { cartItem }, addToCart, cartState, cartDispatch, wishlistState: { wishlistItem }, wishlistDispatch } = useCart();
+    const addToCartHandler = () => {
+        if (isLogin) {
+            cartDispatch({ type: "ADD_TO_CART", payload: data })
+            updateCart(data, token);
+        } else {
+            navigate("/login")
+        }
+    }
+    const addWishlistHandler = () => {
+        if (isLogin) {
+            wishlistDispatch({ type: "ADD_TO_WISHLIST", payload: data })
+            addItemInWishlist(data, token)
+        } else {
+            navigate("/login")
+        }
+    }
+    const removeWishlistHandler = () => {
+        if (isLogin) {
+            wishlistDispatch({ type: "REMOVE_FROM_WISHLIST", payload: data })
+            removeItemFromWishlist(data, token)
+        } else {
+            navigate("/login")
+        }
+    }
+    return (
+        <>
+            <div className="card-vertical">
+                <div className="image-container">
+                    <img
+                        src={data.image}
+                        alt=""
+                        className="card-image"
+                    />
+                    {data.best_selling === "true" && <span className="card-badge"> Best Selling</span>}
```

Comment:
you can also remove 'true' {data.best_selling && <span className="card-badge"> Best Selling</span>}